### PR TITLE
Fix sorting on nested field with unmapped

### LINKED ISSUE #33644: Sorting by unmapped nested field
**Elasticsearch version** (`bin/elasticsearch --version`): 6.4.0

**Plugins installed**: []

**JVM version** (`java -version`): 1.8.0_171

**OS version** (`uname -a` if on a Unix-like system): Windows 10

**Description of the problem including expected versus actual behavior**:

Field `unmapped_type` in **sort** section inside **query** doesn't work properly in case of nested properties. If I query for the data sorted by non-existing (unmapped) nested field I get `query_shard_exception`. The behavior I expect is to sort the data without nested field by a default value of a type specified in `unmapped_type` field. It would be consistent with current behavior for non-nested fields.

**Steps to reproduce**:

Let's say I have 2 indices (One with nested field and one without):

```javascript
PUT index-nested 
{
  "mappings": {
    "doc": {
      "properties": {
        "notNestedField": {
          "type": "keyword"
        },
        "Nested": {
          "type": "nested",
          "properties": {
            "nestedField": {
              "type": "keyword"
            }
          }
        }
      }
    }
  }
}

PUT index-not-nested
{
  "mappings": {
    "doc": {
      "properties": {
        "notNestedField": {
          "type": "keyword"
        }
      }
    }
  }
}
```

The query for sorted data from both indices:

```javascript
GET index-nested,index-not-nested/_search
{
  "sort": [
    {
      "Nested.nestedField": {
        "unmapped_type" : "keyword",
        "nested": {
          "path": "Nested"
        }, 
        "order": "desc"
      }
    }
  ]
}
```

Only data from **index-nested** are returned, all shards belonging to indices without nested field are failing:

```javascript
...
"failures": [
      {
        "shard": 0,
        "index": "index-not-nested",
        "node": "w8_R6Fg8TLGsKq7zCRAXKg",
        "reason": {
          "type": "query_shard_exception",
          "reason": "[nested] failed to find nested object under path [Nested]",
          "index_uuid": "1D0UTBwaRHmnG2TC6PJZdg",
          "index": "index-not-nested"
        }
      }
    ]
...
```

### PR DESCRIPTION
**Pull request by matriv**
Previously sorting on a missing nested field would fail with an
Exception:
`[nested_field] failed to find nested object under path [nested_path]`
despite `unmapped_type` being set on the query.

Fixes: #33644

### CODE REVIEW

**Comment by matriv** [comment]
Tried to add some unit test for this but it's quite difficult because of the mocked `QueryShardContext`. We would need to mock the `getMapperService()` and on the returned `MapperService` the `unmappedFieldType()`, so decided to go only for an integration test.

File: server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java [hunk 1]
```diff
@@ -392,20 +394,22 @@ public SortFieldAndFormat build(QueryShardContext context) throws IOException {
...
-            if (nestedSort != null) {
-                if (context.indexVersionCreated().before(Version.V_6_5_0) && nestedSort.getMaxChildren() != Integer.MAX_VALUE) {
-                    throw new QueryShardException(context,
-                        "max_children is only supported on v6.5.0 or higher");
-                }
-                if (nestedSort.getNestedSort() != null && nestedSort.getMaxChildren() != Integer.MAX_VALUE)  {
-                    throw new QueryShardException(context,
-                        "max_children is only supported on last level of nested sort");
+            Nested nested = null;
+            if (!isUnmapped) {
```
Comment (jimczi) [inline server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java]: nit: we have a convention to use `isUnmapped == false` for readability.

File: server/src/test/java/org/elasticsearch/search/sort/FieldSortIT.java [hunk 2]
```diff
@@ -903,6 +903,14 @@ public void testIgnoreUnmapped() throws Exception {
...
                 .get();
         assertNoFailures(searchResponse);
+
+        // nested field
+        searchResponse = client().prepareSearch()
+            .setQuery(matchAllQuery())
+            .addSort(SortBuilders.fieldSort("nested.foo").unmappedType("keyword")
+                .setNestedSort(new NestedSortBuilder("nested").setNestedSort(new NestedSortBuilder("nested.foo"))))
+            .get();
+        assertNoFailures(searchResponse);
```
Comment (jimczi) [inline server/src/test/java/org/elasticsearch/search/sort/FieldSortIT.java]: Can you add a test with a nested query ?

**Review by jimczi — APPROVED** [review]
Thanks @matriv, I left some minor comments but the change looks good to me.

**Review by cbuescher — APPROVED** [review]
LGTM

**Review by jimczi — APPROVED** [review]
LGTM

**Pull request merged**